impl/src/main/java/org/ehcache/config/serializer/DefaultSerializationProviderConfiguration.java

```diff
@@ -41,40 +42,42 @@
     if (serializerClass == null) {
       throw new NullPointerException("Serializer class cannot be null");
     }
-    String alias = serializableClass.getName();
     
-    if (serializerClass.isAnnotationPresent(Serializer.Transient.class)) {
-      if (serializerClass.isAnnotationPresent(Serializer.Persistent.class)) {
-        if (transientSerializers.containsKey(alias)) {
-          if (persistentSerializers.containsKey(alias)) {
-            throw new IllegalArgumentException("Duplicate transient & persistent serializer for class : " + alias);
-          } else {
-            persistentSerializers.put(alias, serializerClass);
-          }
-        } else if (persistentSerializers.containsKey(alias)) {
-          transientSerializers.put(alias, serializerClass);
-        } else {
-          persistentSerializers.put(alias, serializerClass);
-          transientSerializers.put(alias, serializerClass);
-        }
+    String alias = serializableClass.getName(); 
```

Comment:
Okay, I blame this diff engine for messing up the whole thing. My change set was just this much:

```java
    boolean transientConstructorPresent = false;
    boolean persistentConstructorPresent = false;
    
    if(transientConstructorPresent = isConstructorPresent(serializerClass, ClassLoader.class)) {
      if (transientSerializers.containsKey(alias)) {
        throw new IllegalArgumentException("Duplicate transient serializer for class : " + alias);
      } else {
        transientSerializers.put(alias, serializerClass);
      }
    }
    
    if(persistentConstructorPresent = isConstructorPresent(serializerClass, ClassLoader.class, FileBasedPersistenceContext.class)) {
      if (persistentSerializers.containsKey(alias)) {
        throw new IllegalArgumentException("Duplicate persistent serializer for class : " + alias);
      } else {
        persistentSerializers.put(alias, serializerClass);
      }
    }

    if(!transientConstructorPresent && !persistentConstructorPresent) {
      throw new IllegalArgumentException("The serializer: " + serializerClass.getName()
                                         + " does not meet the constructor requirements for either transient or persistent caches.");
    }
```

And what you are suggesting is this(as it'd simplify the diff set):

```java
    if (isConstructorPresent(serializerClass, ClassLoader.class)) {
      if (isConstructorPresent(serializerClass, ClassLoader.class, FileBasedPersistenceContext.class)) {
        if (transientSerializers.containsKey(alias)) {
          if (persistentSerializers.containsKey(alias)) {
            throw new IllegalArgumentException("Duplicate transient & persistent serializer for class : " + alias);
          } else {
            persistentSerializers.put(alias, serializerClass);
          }
        } else if (persistentSerializers.containsKey(alias)) {
          transientSerializers.put(alias, serializerClass);
        } else {
          persistentSerializers.put(alias, serializerClass);
          transientSerializers.put(alias, serializerClass);
        }
      } else {
        if (transientSerializers.containsKey(alias)) {
            throw new IllegalArgumentException("Duplicate transient serializer for class : " + alias);
        } else {
          transientSerializers.put(alias, serializerClass);
        }
      }
    } else if (isConstructorPresent(serializerClass, ClassLoader.class, FileBasedPersistenceContext.class)) {
      if (persistentSerializers.containsKey(alias)) {
          throw new IllegalArgumentException("Duplicate persistent serializer for class : " + alias);
      } else {
        persistentSerializers.put(alias, serializerClass);
      }
    } else {
      throw new IllegalArgumentException("Serializer class '" + serializerClass + "' does not identify as persistent or transient.");
    }
```

Personally, I feel that the second code block looks too complicated with too many nested if-else blocks, even though both does the same thing. I feel the the first one looks much cleaner. Now, if I get one more veto against my opinion I'll happily change it to the second one :)